### ISSUE TITLE
publish.single("test", []) empty message hangs

### ISSUE THREAD

**Issue by CarlFK** [eclipse/paho.mqtt.python#429]
I haven't seen docs that define what should happen for an empty list.  I would expect this:

>     payload : the payload to be published. If "" or None, a zero length payload
>               will be published.
 - https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/publish.py#L179

instead it gets stuck:

```
import paho.mqtt.publish as publish

publish.single("test", 1) 
publish.single("test", True)
publish.single("test", []) # never returns 
```
until ^c
```
^CTraceback (most recent call last):
  File "t1.py", line 7, in <module>
    publish.single("test", [])
  File "/home/runr/score/venv/lib/python3.7/site-packages/paho/mqtt/publish.py", line 223, in single
    protocol, transport)
  File "/home/runr/score/venv/lib/python3.7/site-packages/paho/mqtt/publish.py", line 160, in multiple
    client.loop_forever()
  File "/home/runr/score/venv/lib/python3.7/site-packages/paho/mqtt/client.py", line 1578, in loop_forever
    rc = self.loop(timeout, max_packets)
  File "/home/runr/score/venv/lib/python3.7/site-packages/paho/mqtt/client.py", line 1057, in loop
    socklist = select.select(rlist, wlist, [], timeout)
KeyboardInterrupt
```

**Comment by PierreF** [comment]
I don't have yet a fix to propose, but the issue is that you send an empty list and not an empty string or None.
So to avoid your issue, use
```
publish.single("test", "[]")  # which send an non-empty payload
publish.single("test", "")  # which send a zero-length payload
```

**Comment by ralight** [comment]
This was addressed in 1.5.1, if you try to send an empty list an exception will be raised.
